db/db_test.cc

```diff
@@ -4054,48 +4054,76 @@ TEST_F(DBTest, ConcurrentFlushWAL) {
   options.env = env_;
   WriteOptions wopt;
   ReadOptions ropt;
-  for (bool two_write_queues : {false, true}) {
-    for (bool manual_wal_flush : {false, true}) {
-      options.two_write_queues = two_write_queues;
-      options.manual_wal_flush = manual_wal_flush;
-      options.create_if_missing = true;
-      DestroyAndReopen(options);
-      std::vector<port::Thread> threads;
-      threads.emplace_back([&] {
-        for (size_t i = 0; i < cnt; i++) {
-          auto istr = ToString(i);
-          db_->Put(wopt, db_->DefaultColumnFamily(), "a" + istr, "b" + istr);
-        }
-      });
-      if (two_write_queues) {
-        threads.emplace_back([&] {
-          for (size_t i = cnt; i < 2 * cnt; i++) {
+  SyncPoint::GetInstance()->SetCallBack("SplitWrite:allowed", [&](void* arg) {
```

Comment:
A general suggestion before diving into the detailed interleaving of the threads:
since this test covers a combination of configurations, it will be worthwhile to define a new parameterized GTest fixture e.g. `DBWALFlushTest` (or something similar) so that we can improve readability.

Reply:
fix it,  add new test ManualFlushWalAndWriteRace